tests/monotouch-test/System.Net.Http/MessageHandlers.cs

```diff
@@ -110,5 +113,51 @@ public void TestNSUrlSessionHandlerCookies ()
 		}
 #endif
 
+		// ensure that if we have a redirect, we do not have the auth headers in the following requests
+		[Test]
+#if !__WATCHOS__
+		[TestCase (typeof (HttpClientHandler))]
+		[TestCase (typeof (CFNetworkHandler))]
+#endif
+		[TestCase (typeof (NSUrlSessionHandler))]
+		public void RedirectionWithAuthorizationHeaders (Type handlerType)
+		{
+			// anonymous type that represent the response
+			var definition = new { 
+				headers = new Dictionary<string, string> (),
+				origin = "",
+				url = ""
+			};
+			
+			PrintHandlerToTest ();
+
+			bool containsAuthorizarion = false;
+			string json = "";
+			bool done = false;
+			Exception ex = null;
+
+			TestRuntime.RunAsync (DateTime.Now.AddSeconds (30), async () =>
+			{
+				try {
+					HttpClient client = new HttpClient (GetHandler (handlerType));
+					client.BaseAddress = new Uri ("https://httpbin.org");
+					var byteArray = new UTF8Encoding ().GetBytes ("username:password");
+					client.DefaultRequestHeaders.Authorization = new AuthenticationHeaderValue ("Basic", Convert.ToBase64String(byteArray));
+					var result = await client.GetAsync ("https://httpbin.org/redirect/3");
+					// get the data returned from httbin which contains the details of the requested performed.
+					json = await result.Content.ReadAsStringAsync ();
+					var httpRequest = JsonConvert.DeserializeAnonymousType (json, definition);
```

Comment:
This is a very simple use case to take a NuGet dependency on, in particular something as heavy as Newtonsoft.Json, which does not work well on the watch (way too big).

There is Json support in the BCL somewhere, it should be possible to use that (or even just a simple "string.Contains" if it's as simple as it looks).